src/main/java/Cubone.java

```diff
@@ -0,0 +1,89 @@
+            String input = sc.nextLine();
+            if (input.equals("bye")) {
+                // exit loop
+                System.out.println(chat_bar + chat_prefix + "Bye. Hope to see you again soon!\n" + chat_bar);
+                break;
+            } else if (input.equals("list")) {
+                // print all user inputs
+                System.out.println(chat_bar + chat_prefix + "Here are the tasks in your list:");
+                for (int i = 0; i < inputed_tasks.size(); i++) {
+                    System.out.println((i + 1) + ". " + inputed_tasks.get(i).toString());
+                }
+                System.out.println(chat_bar);
+            } else if(input.contains("mark") && !input.contains("unmark")){
+                // mark task as done
+                String[] split = input.split(" ");
+                int index = Integer.parseInt(split[1]);
+                inputed_tasks.get(index-1).markAsDone();
+                System.out.println(chat_bar + chat_prefix + "Nice! I've marked this task as done:\n" + inputed_tasks.get(index-1).toString() + "\n" + chat_bar);
+            } else if(input.contains("unmark")){
+                // mark task as undone
+                String[] split = input.split(" ");
+                int index = Integer.parseInt(split[1]);
+                inputed_tasks.get(index-1).markAsUndone();
+                System.out.println(chat_bar + chat_prefix + "OK, I've marked this task as not done yet:\n" + inputed_tasks.get(index-1).toString() + "\n" + chat_bar);
+            } else if (input.contains("todo")){
+                // add todo task
+                String[] split = input.split(" ");
+                if(split.length == 1){
+                    System.out.println(chat_bar + chat_prefix + "☹ OOPS!!! The description of a todo cannot be empty.\n" + chat_bar);
+                }else{
+                    inputed_tasks.add(new Todo(input));
+                    System.out.println(chat_bar + chat_prefix + "Got it. I've added this Todo:\n" + inputed_tasks.get(inputed_tasks.size()-1).toString() + "\n" +
+                                        "now you have " + inputed_tasks.size() + " tasks in the list\n" + chat_bar); 
+                }
+            } else if (input.contains("deadline")){
+                // add deadline task
+                // usage: deadline <description> /by <date>
+                Pattern deadlinePattern = Pattern.compile("deadline (.+?) /by (.+)");
+                Matcher deadlineMatcher = deadlinePattern.matcher(input);
+                if (deadlineMatcher.find()) {
+                    inputed_tasks.add(new Deadline(deadlineMatcher.group(1), deadlineMatcher.group(2)));
+                    System.out.println(chat_bar + chat_prefix + "Got it. I've added this task:\n" + inputed_tasks.get(inputed_tasks.size()-1).toString() + "\n" + 
+                                        "now you have " + inputed_tasks.size() + " tasks in the list\n" + chat_bar);
```

Comment:
The pattern matching and input validation for 'deadline' is slightly complicated, which could be made clearer by splitting the logic into multiple steps or using well-named variables for intermediate results.